### PR TITLE
XCode 12 fixes

### PR DESCRIPTION
**Pull request by MaherKSantina**
- Added tmp.xcconfig which has the configuration to make carthage update command work on XCode 12

## Note
Command `fastlane ios update_dependencies` and `fastlane ios bootstrap` will not work unless you run this command in the terminal first
```
export XCODE_XCCONFIG_FILE=$PWD/tmp.xcconfig
```

I've already configured this in CircleCI so it should work in theory, but I'm going to create this PR to see if the build fails or not

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -46,13 +47,18 @@ jobs:
   build-test:
 
     macos:
-      xcode: 11.7.0
+      xcode: 12.0.1
     environment:
       FL_OUTPUT_DIR: output
       FASTLANE_LANE: test
 
     shell: /bin/bash --login -o pipefail
```
Comment (nickfrollo) [inline .circleci/config.yml:55]: Please remove this line, it caused problems in other projects

File: .circleci/config.yml [hunk 2]
```diff
@@ -62,14 +68,19 @@ jobs:
       - save_cache: *save_carthage_cache
       - run:
           name: Install simulators
-          command: xcrun simctl create 'iPhone 5s (12.4)' com.apple.CoreSimulator.SimDeviceType.iPhone-5s com.apple.CoreSimulator.SimRuntime.iOS-12-4
+          command: xcrun simctl create 'iPhone SE (14.0)' com.apple.CoreSimulator.SimDeviceType.iPhone-SE com.apple.CoreSimulator.SimRuntime.iOS-14-0
```
Comment (nickfrollo) [inline .circleci/config.yml:71]: Can we get a 13.5 sim running? Even if different device

**Review by nickfrollo — CHANGES_REQUESTED** [review]
Good job! Just a couple of tweaks

File: .circleci/config.yml (hunk 2 above)
Comment (MaherKSantina) [reply .circleci/config.yml:71]: Sure thing I'll look for one

File: .circleci/config.yml (hunk 1 above)
Comment (MaherKSantina) [reply .circleci/config.yml:55]: Sure thing

File: Tests/Network/NetworkAuthenticatorTests.swift [hunk 3]
```diff
@@ -369,7 +369,7 @@ class NetworkAuthenticatorTests: XCTestCase {
             expectation3.fulfill()
         }
         
-        wait(for: [expectation1, expectation2, expectation3], timeout: 10.0)
+        wait(for: [expectation1, expectation2, expectation3], timeout: 300.0)
```
Comment (nickfrollo) [inline Tests/Network/NetworkAuthenticatorTests.swift:372]: 5 minutes? Not sure we should wait that long. Instead I'd remove one of the network calls. I'd guess what happens is the 3rd one sometimes fires too late to get cancelled.

**Review by nickfrollo — CHANGES_REQUESTED** [review]
Just one tweak on the tests please! Looking good otherwise!

**Review by nickfrollo — APPROVED** [review]

**Review by sincychacko-frollo — APPROVED** [review]

**Pull request merged**